docs/introduction.md

```diff
@@ -1,40 +1,187 @@
-## What is this? 
+## What's this?
 Envoy Swarm Control Plane is a piece of software that's designed to run inside a Docker Swarm environment.
 It will read swarm service definitions and convert them to configuration objects that Envoy proxies can use.
 
-The goal is to run envoy at the edge as a reverse proxy for your swarm services.
-this means that the focus is on serving external clients rather than facilitating in service-to-service communication. 
+> Envoy Proxy is a modern, high performance, small footprint edge and service proxy. Envoy is most comparable to software load balancers such as NGINX and HAProxy.
+
+source: [Ambassador](https://www.getambassador.io/learn/kubernetes-glossary/envoy-proxy/#what-is-envoy-proxy)
+
+The control plane will configure Envoy to route traffic from the internet towards the many services running on one or more hosts in your swarm. Envoy itself can do much more e.g. handle service-to-service communication,  but with this control plane we'll focus purely on serving external clients.
+
+### Why do I need this?
+Docker swarm does not come with any HTTP routing out of the box. If you are running multiple web services on your swarm you'll quickly find yourself in a conflict as only one service can be published on the HTTP or HTTPS ports. This is a common problem usually solved by introducing a [reverse proxy](https://www.cloudflare.com/learning/cdn/glossary/reverse-proxy/) that takes care of routing traffic and doing other helpful things like terminating TLS.
+
+As you can imagine, these reverse proxies need to be configured. In an environment where addresses change frequently e.g. Docker containers starting and stopping, you might need to reconfigure the proxy multiple times a day. That's something we are automating with this control plane software!
+
+### Why not use X?
+There are many good solutions for this problem already. We'll focus a bit on the differences here.
+// todo: create comparison and link the word "here".
 
 ## Installation
-Below are the steps you should take to set up and run this software. I keep the most up-to-date code examples in the 
+Below are the steps you should take to set up and run this software. I keep the most up-to-date code examples in the
 [deployments](https://github.com/nstapelbroek/envoy-swarm-control-plane/tree/master/deployments) folder of this repository.
 Note that this folder might contain some files and folders that are not relevant for these instructions.
 
-### 1. Setting up your docker swarm 
+### What we'll be configuring
+This control plane works on all kinds of setups regardless if you have 1, 3 or 10+ nodes running with multiple managers. That's one of the cool things! 
+But, to keep things generic, a high-level overview consists of:
+
+1. Envoy instances that run on one or many nodes, which we'll call / mark as `edge` nodes.
```

Comment:
Lets use swarms word of "labeling" your nodes and get rid of calling something tagged or marked. Just to prevent confusion.